### DISCUSSION TITLE
What is the suggestion for managing user details?

### DISCUSSION THREAD

**Discussion by Alik2015** [simpleidserver/SimpleIdServer#655] [Q&A]
When our app is using id server as our auth server.  How do you suggest we allow user to manage their account?  Do we build our own UI allowing them to update details using the id server API or do we just redirect them to the id server where they can manage their account?

Both options are good but using our own UI would enable us to manage other details of our system, but it will mean that all functions would need to be implemented on our side using the API.

I am wondering how people have achieved account management for their apps.

**Comment by simpleidserver** [comment]
I don't have a preference, but it is possible to edit the account information in two different ways.

The first approach involves providing the URL of the Identity Server and allowing the user to manage their local information, such as credentials and external accounts. 
Views can be customized by editing the CSHTML files, such as the Layout (/Views/_Layout.cshtml) and the Profile (/Home/Profile.cshtml).
If you need a custom action, such as updating personal information like the first name, you can create a new  `CustomHomeController`  controller and inherit from the `HomeController`.

Example of a custom controller.

```
public class CustomHomeController : HomeController
{
    public CustomHomeController(
        IOptions<IdServerHostOptions> options, 
        IUserHelper userHelper, 
        IUserRepository userRepository, 
        IClientRepository clientRepository, 
        IUmaPendingRequestRepository pendingRequestRepository, 
        IAuthenticationSchemeProvider authenticationSchemeProvider, 
        IOTPQRCodeGenerator otpQRCodeGenerator, 
        ICredentialTemplateRepository credentialTemplateRepository, 
        IBusControl busControl, 
        IEnumerable<IAuthenticationMethodService> authenticationMethodServices, 
        IUserSessionResitory userSessionRepository, 
        IRecurringJobManager recurringJobManager, 
        ILogger<HomeController> logger) : base(options, userHelper, userRepository, clientRepository, pendingRequestRepository, authenticationSchemeProvider, otpQRCodeGenerator, credentialTemplateRepository, busControl, authenticationMethodServices, userSessionRepository, recurringJobManager, logger)
    {
    }

    [HttpGet]
    [Authorize(Constants.Policies.Authenticated)]
    public override async Task<IActionResult> Profile([FromRoute] string prefix, CancellationToken cancellationToken)
    {
        prefix = prefix ?? Constants.DefaultRealm;
        var result = new CustomProfileViewModel();
        await base.Build(prefix, result, cancellationToken);
        return View(result);
    }
	
	[HttpPost]
    public void UpdatePersonalInformation()
    {

    }
}
```

Once the custom controller is created, edit the `Program.cs` file to add a custom route. 
It will redirect URL /Home/Profile to your custom controller. Don't forget to rename the `Views/Home` folder to `Views/CustomHome`.

```
app.MapControllerRoute("customHomeController",
    pattern: (identityServerConfiguration.IsRealmEnabled ? "{prefix}/" : string.Empty) + "Home/{action=Index}/{id?}",
    defaults: new { controller = "CustomHome" });
	
app.UseSID()
etc...
```

The second approach involves developing a custom web application and calling our different API operations to update account information, such as credentials. 
A list of available operations is provided in the Swagger endpoint :)
